internals/secrethub/tree_test.go

```diff
@@ -0,0 +1,237 @@
+				"│   └── test/repo/secretFolder/found you\n" +
+				"└── test/repo/mySecret\n\n" +
+				"1 directory, 2 secrets\n",
+		},
+		"no indent": {
+			cmd: &TreeCommand{
+				io:            ui.NewUserIO(),
+				noIndentation: true,
+				newClient: func() (secrethub.ClientInterface, error) {
+					return &secrethub.Client{}, nil
+				},
+			},
+			expectedOutput: "test/repo/\n" +
+				"secretFolder/\n" +
+				"found you\n" +
+				"mySecret\n\n" +
+				"1 directory, 2 secrets\n",
+		},
+		"no report": {
+			cmd: &TreeCommand{
+				io:       ui.NewUserIO(),
+				noReport: true,
+				newClient: func() (secrethub.ClientInterface, error) {
+					return &secrethub.Client{}, nil
+				},
+			},
+			expectedOutput: "test/repo/\n" +
+				"├── secretFolder/\n" +
+				"│   └── found you\n" +
+				"└── mySecret\n",
+		},
+		"all flags": {
+			cmd: &TreeCommand{
+				path:          "test/repo",
+				io:            ui.NewUserIO(),
+				fullPaths:     true,
+				noIndentation: true,
+				noReport:      true,
+				newClient: func() (secrethub.ClientInterface, error) {
+					return &secrethub.Client{}, nil
+				},
+			},
+			expectedOutput: "test/repo/\n" +
+				"test/repo/secretFolder/\n" +
+				"test/repo/secretFolder/found you\n" +
+				"test/repo/mySecret\n",
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			w := &bytes.Buffer{}
+			tc.cmd.printTree(tree, w)
+			assert.Equal(t, w.String(), tc.expectedOutput)
+		})
+	}
+}
+
+func TestTreeColoring(t *testing.T) {
+	uuid0, _ := uuid.FromString("0")
+	uuid1, _ := uuid.FromString("1")
+	uuid2, _ := uuid.FromString("2")
+	tree := &api.Tree{
+		RootDir: &api.Dir{
+			Name:   "test/repo",
+			Status: "flagged",
+			DirID:  uuid0,
+			SubDirs: []*api.Dir{
+				{
+					Name:     "happy",
+					DirID:    uuid1,
+					ParentID: &uuid0,
+				},
+				{
+					Name:     "secretFolder",
+					DirID:    uuid2,
+					ParentID: &uuid0,
+					Status:   "flagged",
+					Secrets: []*api.Secret{
+						{
+							Name:   "found you",
+							Status: "flagged",
+						},
+					},
+				},
+			},
+			Secrets: []*api.Secret{
+				{Name: "mySecret"},
+			},
+		},
+		Dirs: map[uuid.UUID]*api.Dir{
+			uuid.New(): {
+				DirID:    uuid0,
+				ParentID: &uuid0,
+			},
+			uuid.New(): {
+				Name:     "happy",
+				DirID:    uuid1,
+				ParentID: &uuid0,
+			},
+			uuid.New(): {
+				Name:     "secretFolder",
+				DirID:    uuid2,
+				ParentID: &uuid0,
+				Secrets: []*api.Secret{
+					{Name: "found you"},
+				},
+			},
+		},
+		Secrets: map[uuid.UUID]*api.Secret{
+			uuid.New(): {Name: "found you"},
+			uuid.New(): {Name: "mySecret"},
+		},
+	}
+	cases := map[string]struct {
+		cmd            *TreeCommand
+		expectedOutput string
+	}{
+		"simple tree": {
+			cmd: &TreeCommand{
+				newClient: func() (secrethub.ClientInterface, error) {
+					return &secrethub.Client{}, nil
+				},
+			},
+			expectedOutput: "test/repo/\n" +
+				"├── happy/\n" +
+				"├── secretFolder/\n" +
+				"│   └── found you\n" +
```

Comment:
We'd expect this one to be red, right?

Reply:
Indeed. The tests have been adjusted to work properly